pkg/providers/cloudstack/validator.go

```diff
@@ -234,29 +280,24 @@ func (v *Validator) validateMachineConfig(ctx context.Context, datacenterConfig
 			return fmt.Errorf("restricted key %s found in custom user details", restrictedKey)
 		}
 	}
-	zones, err := v.cmk.ValidateZonesPresent(ctx, datacenterConfig.Spec.Zones)
-	if err != nil {
-		return fmt.Errorf("checking zones %v", err)
-	}
-	account := datacenterConfig.Spec.Account
 
-	for _, zone := range zones {
-		if err = v.cmk.ValidateTemplatePresent(ctx, domainId, zone.Id, account, machineConfig.Spec.Template); err != nil {
+	for _, az := range v.availabilityZones {
+		cmk := v.cmks[az.Name]
```

Comment:
I think it would be good to check that the cmk exists in the map again

Reply:
Done